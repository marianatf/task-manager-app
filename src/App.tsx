import { TaskList } from "./components/TaskList";
import { useTasks } from "./hooks/useTasks";

const App = () => {
  const {
    activeCount,
    completedCount,
    handleClearAllCompleted,
    handleCompleted,
    handleDelete,
    handleCreate,
    tasks,
  } = useTasks();

  return (
    <TaskList
      tasks={tasks}
      onDelete={handleDelete}
      onToggleComplete={handleCompleted}
      activeCount={activeCount}
      completedCount={completedCount}
      onClearCompleted={handleClearAllCompleted}
      onCreateTask={handleCreate}
    />
  );
};

export default App;
