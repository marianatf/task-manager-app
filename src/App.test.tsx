import { render } from "@testing-library/react";
import App from "./App";

import { createRoot } from "react-dom/client";
import { describe, expect, it } from "vitest";

describe("App", () => {
  it("should match snapshot", () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders without crashing", () => {
    const div = document.createElement("div");
    const root = createRoot(div);
    root.render(<App />);
    root.unmount();
  });

  it("renders taskList correctly", () => {
  const { getByText } = render(<App />);
  expect(getByText(/to-do list/i)).toBeInTheDocument();
});
});
