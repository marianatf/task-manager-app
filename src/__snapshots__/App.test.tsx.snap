// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex items-center justify-center min-h-screen bg-gray-100 p-4"
  >
    <div
      class="w-full max-w-md bg-white rounded-lg shadow-md border border-gray-200"
    >
      <header>
        <h1
          class="text-4xl font-bold text-center my-4"
        >
          To-Do List
        </h1>
        <form
          class="flex flex-col gap-3 m-4"
        >
          <input
            class="rounded-lg px-4 py-2 border border-gray-300 focus:outline-none focus:ring-2 focus:ring-black"
            placeholder="Add a task"
            value=""
          />
        </form>
      </header>
      <ul
        class="w-full divide-y divide-gray-200"
      />
      <footer
        class="flex justify-between items-center px-4 py-2 border-t border-gray-200 text-gray-600 w-full"
      >
        <span>
          <strong
            class="font-semibold"
          >
            0
          </strong>
           outstanding tasks
        </span>
      </footer>
    </div>
  </div>
</DocumentFragment>
`;
